src/main/java/com/cloudbees/jenkins/plugins/bitbucket/JsonParser.java

```diff
@@ -23,25 +23,58 @@
  */
 package com.cloudbees.jenkins.plugins.bitbucket;
 
-import com.google.common.base.Charsets;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.io.Reader;
 import java.io.StringReader;
-import java.text.SimpleDateFormat;
-import java.util.TimeZone;
+import java.text.ParsePosition;
+import java.util.Date;
+
 import org.codehaus.jackson.map.DeserializationConfig;
 import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.map.util.ISO8601DateFormat;
+import org.codehaus.jackson.map.util.ISO8601Utils;
 import org.kohsuke.accmod.Restricted;
 import org.kohsuke.accmod.restrictions.NoExternalUse;
 
+import com.google.common.base.Charsets;
+
 /**
  * Jackson based JSON parser
  */
 @Restricted(NoExternalUse.class)
 public final class JsonParser {
-    private static final String DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSS";
+    /*package*/ static class BitbucketDateFormat extends ISO8601DateFormat {
```

Comment:
Might need to change this after updating Jackson 😇

Reply:
Actually I think you can skip this function and just use `StdDateFormat` from Jackson which support the case your custom parser is trying to fix. The StdDateFormat allows upto 9 milliseconds anything longer it complains which I would too 😅

Reply:
Ok your right, with new Jackson version from master the StdDateFormat works correctly also in the above cases (ok)